src/CoreMedia/CMSampleBuffer.cs

```diff
@@ -834,6 +841,70 @@ public CMSampleBufferError CopyPCMDataIntoAudioBufferList (int frameOffset, int
 
 			return new CMSampleBuffer (buffer, true);
 		}
+
+		[iOS (12,2)]
+		[TV (12,2)]
+		[Mac (10,14,4, onlyOn64: true)]
+		[DllImport (Constants.CoreMediaLibrary)]
+		static unsafe extern /* OSStatus */ CMSampleBufferError CMSampleBufferCreateForImageBufferWithMakeDataReadyHandler (
+			/* CFAllocatorRef CM_NULLABLE */ IntPtr allocator,
+			/* CVImageBufferRef CM_NONNULL */ IntPtr imageBuffer,
+			/* Boolean */ [MarshalAs (UnmanagedType.I1)] bool dataReady,
+			/* CMSampleBufferMakeDataReadyHandler CM_NULLABLE */ IntPtr makeDataReadyHandler,
+			/* CMVideoFormatDescriptionRef CM_NONNULL */ IntPtr formatDescription,
+			/* const CMSampleTimingInfo* CM_NONNULL */ CMSampleTimingInfo [] sampleTiming,
+			/* CM_RETURNS_RETAINED_PARAMETER CMSampleBufferRef CM_NULLABLE * CM_NONNULL */ out IntPtr sampleBufferOut);
+
+		internal delegate CMSampleBufferError DCMSampleBufferMakeDataReadyHandler (IntPtr block, IntPtr sbuf);
+
+		static internal class DCMSampleBufferMakeDataReadyHandlerTrampoline {
+			static internal readonly DCMSampleBufferMakeDataReadyHandler Handler = Invoke;
+
+			[MonoPInvokeCallback (typeof (DCMSampleBufferMakeDataReadyHandler))]
+			static unsafe CMSampleBufferError Invoke (IntPtr block, IntPtr sbuf)
+			{
+				var descriptor = (BlockLiteral *) block;
+				var del = (CMSampleBufferMakeDataReadyHandler) descriptor->Target;
+				return del (new CMSampleBuffer (sbuf));
```

Comment:
I wonder if we should do something like this instead

```csharp
using (var b = new CMSampleBuffer (sbuf)) {
	return del (b);
}
```

You can eventually run out of sample buffers because the system tries to be clever and reuse them and our GC is not fast enough to Dispose them so it needs a little help by disposing it right after it is not used anymore. This has got some of our users in the past.